### ISSUE TITLE
Reusing classes with pseudo-selectors don't build

### ISSUE THREAD

**Issue by kamilmccabe** [maizzle/framework#852]
- Maizzle Version: 4.3.1
- Node.js Version: 16.15.0

In the upgrade to v4 from v3 I have discovered an issue with custom styles/classes and pseudo-selectors (namely `hover:` and `dark:`). Previously we had classes written as such
```
.primary-text-color {
    @apply text-grey-10;
    @apply dark:text-grey-80 !important;
  }

.link {
    @apply border-b border-coral border-solid border-t-0 border-l-0 border-r-0;
    @apply hover:text-coral !important;
    text-decoration: none;
  }
```
and in v3 this was working correctly.

Since moving to v4 this hasn't worked and the CSS above isn't being compiled and added to the build file, and the HTML elements that use these classes don't have the classes on them.

To workaround this we can do the following, which works
```
.primary-text-color {
    @apply text-grey-10;
  }
.primary-text-color-dark {
    @apply text-grey-80 !important;
  }
```
and in the HTML
```
<p class="primary-text-color dark:primary-text-color-dark">example</p>
```
The same goes for `hover:`.

Another workaround is to remove the `inlineCss: true` config (then it all works as expected too, however neither of these solutions are ideal).

Is there possibly some extra config I am missing for v4 that is preventing this from building correctly?

**Comment by cossssmin** [comment]
Hmm just tried that in a fresh project and it works fine:

```postcss
/* src/css/utilities.css */

.primary-text-color {
  @apply text-gray-100;
  @apply dark:text-gray-800 !important;
}
```

```html
<extends src="src/layouts/main.html">
  <block name="template">
    <div class="primary-text-color">test</div>
  </block>
</extends>
```

The `<style>` tag in the compiled HTML looks like this:

```html
<html>
<!-- ... -->

<head>
	<style>
	  @media (prefers-color-scheme: dark) {
	    .primary-text-color {
	      color: #1f2937 !important;
	    }
	  }
	  :root {
	    color-scheme: light dark;
	  }
	</style>
</head>
<body>
  <!-- ... -->

  <div style="color: #f3f4f6">test</div>
</body>
</html>
```

Try checking your config/tailwind.config against the [Starter ones](https://github.com/maizzle/maizzle), eventually do a [clean install](https://maizzle.com/docs/installation#clean-update). If that doesn't help, please provide a repository that reproduces the issue so I can help.

**Comment by cossssmin** [comment]
Btw I suppose `*-grey` are your custom colors in your `tailwind.config.js`, but just in case wanted to mention for anyone else seeing this that those no longer exist in Tailwind CSS, they've moved from `grey` to `gray` a long time ago. So if you're not defining those yourself, that may also be an issue.

**Comment by kamilmccabe** [comment]
Sorry yes `grey` is defined ourselves. I can't see anything in the config or tailwind.config that looks wrong. Does yours work when you have `inlineCss: true` in config?

**Comment by cossssmin** [comment]
> Does yours work when you have inlineCss: true in config?

Yes, as you can see from the sample output above, the color is inlined on the div.

**Comment by kamilmccabe** [comment]
So I have tested it with a fresh project and when I run `maizzle serve` it works, however when I run `maizzle serve production` it doesn't. I think this is to do with the `inlineCss` config, however I believe this used to work in v3.

**Comment by cossssmin** [comment]
I'm sorry, I was tired, it's obvious it can't work, the `div` no longer has the class 🤦‍♂️

Looks like it's caused by [removeInlinedSelectors.js](https://github.com/maizzle/framework/blob/master/src/transformers/removeInlinedSelectors.js) that was introduced in v4 to help preserve selectors that do not exist in HTML, like those for email client targeting.

I'll have a look at what can be done, it should skip media queries and such - sorry about that!

Meanwhile if you don't use email client targeting utilities in your project, you can enable the 'old' way like this:

```js
module.exports = {
  // ...
  inlineCSS: {
    removeStyleTags: true,
  },
  removeInlinedClasses: false,
  removeUnusedCSS: true,
}
```

**Comment by kamilmccabe** [comment]
No worries! Yeah like I mentioned at the start we have implemented the workaround by using multiple classes, but let me know if gets fixed and we can revert it. Thank you for your help!

**Comment by cossssmin** [comment]
Published https://github.com/maizzle/framework/releases/tag/v4.4.0-beta.3, give it a try.

You'll need to install it from the `next` tag: `npm install @maizzle/framework@next`. 

Best do a clean install just to be sure.

**Comment by cossssmin** [comment]
Noticed it doesn't remove some classes when you have `shorthandInlineCSS: true`, will look into it...

**Comment by kamilmccabe** [comment]
I've just tried with `v4.4.0-beta.3` and it seems like no css is being applied now. I can see the classes are being added to the html but there are no associated styles 😕

**Comment by cossssmin** [comment]
Hmm I'm only seeing the `shorthandInlineCSS` issue that I mentioned and I think that may be solved by running `removeUnsedCSS` after it (now it runs before, and when it tries to clean up the inline styles have been merged and no longer fit what's in the head, thus the issue...). 

Would it be possible for you to share a repo that reproduces the issues you're having? Doesn't need to contain everything in your project, just one template and your css/configs so I can understand where it's coming from. Can be a private repo, whatever works for you, I'd really appreciate it 👍

**Comment by cossssmin** [comment]
Released https://github.com/maizzle/framework/releases/tag/v4.4.0-beta.4

Tested with a fresh project, including stuff like `@apply` that we discussed, and it seems to be working as expected.

Again, if you're still seeing issues, I'd be more than happy to look at a repo :)

**Comment by kamilmccabe** [comment]
Thank you for sorting, think it works fine I just had an error in my config. :)

**Comment by kamilmccabe** [comment]
Hi, sorry I haven't manage to take a proper look at that yet. We actually were on `v4.0.2` when I upgraded to this beta to try the fix, so I'm not sure if I've missed something between those upgrade with the config. I will have to take a look when I have time and see what the issue is, but the starter repo seems to work with the correct CSS now.
